File: sample-v7/src/main/java/io/erva/sample/single/SingleChoiceCell.java
package io.erva.sample.single;

import android.view.View;
import android.widget.RadioButton;

import butterknife.BindView;
import butterknife.OnClick;
import io.erva.celladapter.Layout;
import io.erva.celladapter.v7.Cell;
import io.erva.celladapter.v7.select.mode.SelectionManager;
import io.erva.sample.BaseSelectableCell;
import io.erva.sample.R;

@Layout(R.layout.item_single)
public class SingleChoiceCell extends BaseSelectableCell<SingleChoiceModel, Cell.Listener<SingleChoiceModel>> {

    @BindView(R.id.rb_single) RadioButton radioButton;

    public SingleChoiceCell(View view) {
        super(view);
    }

    @Override
    protected void bindView() {
        radioButton.setText(getItem().singleTitle);
        radioButton.setChecked(selectionManager.isSelected(getAdapterPosition()));
    }

    @Override
    public void setSelectionManager(SelectionManager selectionManager) {
        this.selectionManager = selectionManager;
    }

    @OnClick(R.id.rb_single)
    void radioButtonClicked() {
        selectionManager.toggleSelection(getAdapterPosition());
        getListener().onCellClicked(getItem());
    }
}

File: sample-kotlin/src/main/kotlin/io/erva/sample/single/SingleChoiceCell.kt
package io.erva.sample.single

import android.view.View
import io.erva.celladapter.Cell
import io.erva.celladapter.Layout
import io.erva.celladapter.select.SelectableCell
import io.erva.sample.R
import kotlinx.android.synthetic.main.item_single.view.*

@Layout(R.layout.item_single)
class SingleChoiceCell(view: View) : SelectableCell<SingleChoiceModel, Cell.Listener<SingleChoiceModel>>(view) {

    override fun bindView() {
        view.rb_single.text = item().singleTitle
        view.rb_single.isChecked = selectionManager.isSelected(adapterPosition)
        view.rb_single.setOnClickListener {
            selectionManager.toggleSelection(adapterPosition)
            listener()?.onCellClicked(item())
        }
    }
}